micropython/umqtt.simple/umqtt/simple.py

```diff
@@ -60,8 +60,16 @@ def set_last_will(self, topic, msg, retain=False, qos=0):
         self.lw_qos = qos
         self.lw_retain = retain
 
+    def settimeout(self, value):
+        '''Blocking socket can cause non determinstic hang incase of n/w failure.
+        Hence, by setting timeout, application programmer can get exception in case of failure.
+        This enable him/her to take appropriate action like restarting the task etc.
+        This works for async apps too.
+        '''
+        self._sock.settimeout(value)
+
     def connect(self, clean_session=True):
-        self.sock = socket.socket()
+        self._sock = self.sock = socket.socket()
```

Comment:
I think it would be better to instead add a `timeout=None` argument to this `connect()` method.  Then call `self.sock.settimeout(timeout)` after creating the socket.

That would mean that the connect call below, and wrap_socket, would also apply the timeout.

Reply:
Hi @dpgeorge ,
Agree with you. That will be better.

Reply:
@prabhu-yu Will you update this PR with the above suggestion?  If not I can do it.

Reply:
Hi @dpgeorge,
I will do. Please give me a day. 
I will test and get back to you with changes.